framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -194,62 +171,138 @@ DerivativeMaterialInterface<T>::getDefaultMaterialProperty(const std::string & p
   return _zero;
 }
 
+template<>
+template<typename U>
+bool
+DerivativeMaterialInterface<Material>::haveMaterialProperty(const std::string & prop_name)
+{
+  return ((this->boundaryRestricted() && this->template hasBoundaryMaterialProperty<U>(prop_name)) ||
+         (this->template hasBlockMaterialProperty<U>(prop_name)));
+}
+
 template<class T>
 template<typename U>
-MaterialProperty<U> &
-DerivativeMaterialInterface<T>::declarePropertyDerivative(const std::string &base, const std::vector<std::string> &c)
+bool
+DerivativeMaterialInterface<T>::haveMaterialProperty(const std::string & prop_name)
 {
-  return this->template declareProperty<U>(propertyName(base, c));
+  // Call the correct method to test for material property declarations
+  BlockRestrictable * blk = dynamic_cast<BlockRestrictable *>(this);
+  BoundaryRestrictable * bnd = dynamic_cast<BoundaryRestrictable *>(this);
```

Comment:
This is code that @aeslaughter added and the cast results **are** checked for NULL (```bnd &&``` in the if).